crates/metastore/src/client.rs

```diff
@@ -323,6 +323,36 @@ impl StatefulWorker {
                     error!("failed to respond to ping");
                 }
             }
+            ClientRequest::Commit {
+                version: current_catalog_version,
+                state,
+                response,
+            } => {
+                let state: CatalogState = state.as_ref().clone();
+                if state.version <= current_catalog_version {
+                    let e =  Err(CatalogError::new(format!(
+                        "cannot commit outdated state: state version: {}, current catalog version: {}",
+                        state.version, current_catalog_version
+                    )));
+                    return response.send(e).unwrap();
+                }
+
+                let commit_request = tonic::Request::new(CommitRequest {
+                    db_id: self.db_id.into_bytes().to_vec(),
+                    catalog_version: current_catalog_version,
+                    catalog: Some(state.try_into().unwrap()),
+                });
+
+                let result = self.client.commit_catalog(commit_request).await.unwrap();
+                let result = result.into_inner();
+                let state: CatalogState = result.catalog.unwrap().try_into().unwrap();
```

Comment:
are these unwraps safe?

Reply:
If these ever do panic, they won't crash the application as they're over the grpc client which has some built in panic handling.

Reply:
our other endpoints are also unwrapping `client.ping().await.unwrap();`